forgerock-authenticator/src/main/java/org/forgerock/android/auth/FRAPolicyEvaluator.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2023 ForgeRock. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+package org.forgerock.android.auth;
+
+import android.content.Context;
+
+import androidx.annotation.NonNull;
+
+import org.forgerock.android.auth.policy.FRAPolicy;
+import org.forgerock.android.auth.policy.BiometricAvailablePolicy;
+import org.forgerock.android.auth.policy.DeviceTamperingPolicy;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * The Policy Evaluator is used by the SDK to enforce Policy rules, such as Device Tampering Policy.
+ * It consist of one or more {@link FRAPolicy} objects. Each Policy contain instructions that
+ * determine whether it comply to a particular condition at a particular time.
+ */
+public class FRAPolicyEvaluator {
+
+    private final List<FRAPolicy> policies;
+
+    private static final String TAG = FRAPolicyEvaluator.class.getSimpleName();
+
+    public static final List<FRAPolicy> DEFAULT_POLICIES = new ArrayList<>();
+
+    static {
+        DEFAULT_POLICIES.add(new DeviceTamperingPolicy());
+        DEFAULT_POLICIES.add(new BiometricAvailablePolicy());
+    }
+
+    private FRAPolicy nonCompliancePolicy;
+    private List<FRAPolicy> targetedPolicies = new ArrayList<>();
+
+    private FRAPolicyEvaluator(List<FRAPolicy> policies) {
+        this.policies = policies;
+    }
+
+    /**
+     * Obtain FRAPolicyEvaluator builder. Policies can be configured in the builder, like setting a
+     * new custom Policy.
+     * @return FRAPolicyEvaluatorBuilder the builder to initialize the FRAClient
+     * */
+    public static FRAPolicyEvaluator.FRAPolicyEvaluatorBuilder builder() {
+        return new FRAPolicyEvaluator.FRAPolicyEvaluatorBuilder();
+    }
+
+    /**
+     * The asynchronous PolicyEvaluator builder.
+     */
+    public static class FRAPolicyEvaluatorBuilder {
+        private List<FRAPolicy> policies;
+
+        /** Sets a custom list of policies.
+         * @param policies The list with custom policies.
+         * @return this builder
+         */
+        public FRAPolicyEvaluatorBuilder withPolicies(@NonNull List<FRAPolicy> policies) {
```

Comment:
It will be good if we have withPolicy(FRAPolicy policy), so developer can do 
FRAPolicyEvaluatorBuilder().withPolicy(BiometricPolicy()).withPolicy(DeviceTemperingPolicy()).build()

Reply:
Similar pattern is using under org.forgerock.android.auth.AuthService#AuthService, but I am using lombok, with @Builder and @Singluar, you can delombok it and see how they generate the inteface.

Reply:
I like the idea. Lets see if we can add it easily

Reply:
Done

Reply:
Looks like it will replace the existing list with withPolicies(), should it add to existing list?
FRAPolicyEvaluatorBuilder.withPolicy(A).withPolicy(B).withPolicy([C,D]), A and B will be removed.

Reply:
Yes, that's what is the javadoc of withPolicies `Sets a list of policies to be evaluated. This method overwrite an existing list.`

So, the developer should use something like this instead: 

```
FRAPolicyEvaluator.builder()
    .withPolicies([C,D])
    .withPolicy(A)
    .withPolicy(B)
```